protocol/proofs/new-chains.md

```diff
@@ -0,0 +1,70 @@
+# Purpose
+
+To identify the challenges with starting a new chain using fault proofs and select an option to make the process simple
+and reliable.
+
+# Summary
+
+New chains are recommended to use the `contracts/v1.6.0` release, which includes fault proofs. While this works better
+with deployment tools, cyclic dependencies between the L1 contract deployments and the L2 genesis generation make it
+impossible to create a fully valid deployment context with the `faultGameAbsolutePrestate` and
+`faultGameGenesisOutputRoot` values being unknowable.
+
+# Problem Statement + Context
+
+The current chain deployment process involves deploying all the L1 contracts, then generating the L2 genesis state.
+This is required because the L2 genesis state references a number of L1 contracts, including the `SystemConfig` and
+`OptimismPortal`. The fault proof system creates two cyclic dependencies:
+
+1. The `faultGameGenesisOutputRoot` is used by the L1 `AnchorStateRegistry` contract as the initial anchor state for
+   permissioned and permissionless games. For new chains it must be set to the L2 genesis output root, which is
+   unknowable prior to deploing the L1 contracts.
+2. The `faultGameAbsolutePrestate` is used by the L1 `FaultDisputeGame` contract to define the version of `op-program`
+   to be used as part of permissioned and permissionless dispute games. `op-program` requires the L2 chain
+   configuration (rollup config and L2 genesis state) to be available in the version of the superchain-registry that it
+   includes to operate correctly. Similar to above, the L2 chain configuration is unknowable prior to deploying the L1
+   contracts. Additionally, chains cannot be added to the superchain-registry until _after_ they have launched.
+
+# Proposed Solution
+
+TBD
+
+# Alternatives Considered
```

Comment:
Trying to brainstorm other ideas. I don't know enough about the implementation details to know if these are feasible, but:
1. Would an MCP L2 solve the problem? In an MCP L2 world, all code at all address in genesis becomes identical, but storage will still vary because of chain-specific L1 addresses that live in predeploy storage. Because of the storage differences at genesis I suspect this solution will not work.
    1. What if the genesis was 100% identical? This would be possible if storage values were pushed from L1 to L2 in the first block via an L1 attributes transaction, though I know @tynes wants to minimize the amount of data we pass that way, and this increases it.
3. What about something like: "If `faultGameGenesisOutputRoot` and `faultGameAbsolutePrestate` are both zero, ask the node for the ouput root and compute the prestate". In other words, get these values on-demand the first time a game is played

Reply:
1. If the L2 genesis was always identical I think that would help. The superchain-registry part of this I think would still be an issue but that would be more administrative without having a fundamental cyclic dependency. Not sure if we can make the L2 genesis identical for all new chains though - seems like quite a challenge.
2. The game is created on the L1 chain so it can't reach out to a node for data. `faultGameGenesisOutputRoot` and `faultGameAbsolutePrestate` are the key sources of truth the fault system is rooted in and they aren't available on-chain otherwise.
3. I suspect this gets a lot worse post-interop. You'd need the L2 genesis of all chains in the dependency set. And adding a new chain into the dependency set would be an interesting challenge too - as would migrating an existing chain to be part of the shared DisputeGameFactory (and migrating it out again if needed). There's quite a lot of open questions there to be honest.

Reply:
> If the L2 genesis was always identical I think that would help. The superchain-registry part of this I think would still be an issue but that would be more administrative without having a fundamental cyclic dependency. Not sure if we can make the L2 genesis identical for all new chains though - seems like quite a challenge.

We are planning to have identical L2 genesis as part of Isthmus in Cycle 33, dates [here](https://discord.com/channels/1244729134312198194/1290809960732692534/1291317994177433673) (https://github.com/ethereum-optimism/specs/discussions/373#discussioncomment-10814320). This would remove the fundamental cyclic dependency. 

Do we have a need for deploying chains straight to permissionless before then? If not, we can probably close this design doc, and work out the superchain registry sequencing separately

Reply:
Yeah I think that solves the fundamental issue.  I'm happy to close this in the meantime then. Honestly I just miss having a place I could put a doc for discussion rather than having to put it through an approval process. This PR will do for that. :)